addons/pos_restaurant/static/src/app/screens/floor_screen/floor_screen.js

```diff
@@ -530,6 +535,25 @@ export class FloorScreen extends Component {
         const table = await this.pos.data.create("restaurant.table", [newTableData]);
         return table[0];
     }
+    async unMergeTable(table) {
+        const destinationTable = this.pos.getMainTable(table);
+        const destinationTableOrder = this.pos.getActiveOrdersOnTable(destinationTable)?.[0];
+        const orderToRestore = this.pos.models["pos.order"].filter(
+            (o) =>
+                (o.table_id?.id === table.id || o.uiState.initialTable === table.id) && !o.finalized
+        )[0];
```

Comment:
```suggestion
        const orderToRestore = table["<-pos.order.table_id"].filter((o) => !o.finalized)[0];
```

Reply:
What if destination table was empty while merging so now we will get empty value for `table["<-pos.order.table_id"]`
to handle this we need to use `table.parent_id` in while loop because there may be scenerio
like: table 1(with orderline) merged to table 2(empty) merged to table 3(empty)

in that case 
```
const orderToRestore = this.pos.models["pos.order"].find(
            (o) =>
                (o.table_id?.id === table.id || o.uiState.initialTable === table.id) && !o.finalized
        )[0];
 ```
        
  we can use this

Reply:
If the table being unmerged was originally empty, I think you should do nothing, right?

Reply:
If the main table is empty, are you moving the order (from secondary table) to the main table?

Reply:
Nope the table being unmerged was has an order but the destination table was empty 
in that case, while merging the the table_id of that order will updated to destination table 

so while unmerge we need to again update that order's table_id to original one

see: https://www.awesomescreenshot.com/video/34247039?key=6fb10e6df5e46e6f1f80d4756abf8b81